### PR TITLE
Ft get buyer cart

### PR DESCRIPTION
**Pull request by aapetsi**
## Description
Describe change or new feature here.
## Type of change
Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update
## Checklist
Remove any items which are not applicable.
- [x] I have performed a self-review of my own code
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works .
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: controllers/cart/getCart.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+const Cart = require('../../models/cart')
+const Store = require('../../models/store')
+const Product = require('../../models/product')
+
+async function getCart(req, res) {
+  try {
+    const storeId = req.params.store_id
```
Comment (Rexben001) [inline controllers/cart/getCart.js]: I don't think we'll be supplying any storeId, we'll be supplying only the cartId

File: controllers/cart/getCart.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          as: 'productObjects'
+        }
+      }
+    ])
+
+    const storeCart = populatedCart.filter(
+      item => String(item.storeId) === String(storeId)
+    )
+
+    res.status(200).json(storeCart)
```
Comment (Rexben001) [inline controllers/cart/getCart.js:32]: Will this return something like this
```
   {
  contents: [
    { id: 6789087343, price: 400, description: "Fine product", name: "Rice" },
    { id: 6789087343, price: 400, description: "Fine product", name: "Rice" }
  ],
  checkedOut: false,
  _id: "5e1df5f957aff215deecc4b7",
  agreedPrice: 45,
  total: 540,
  storeId: "5e1def4257aff215deecc4ac",
  checkoutDate: "2020-01-14T17:10:17.173Z",
  __v: 0
};
```

File: routes/cartRouter.js [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
 
 // @route POST api/store/:store_id/cart
 // @desc  Add items to cart
 // @access Public
 router.post('/:store_id/cart', addCart)
 
+// @route GET api/store/:store_id/
+// @desc  Get items from cart
+// @access Public
+router.get('/:store_id/cart', getCart)
```
Comment (Rexben001) [inline routes/cartRouter.js]: Replace with `/store/:cartId`

**Review by Rexben001 — CHANGES_REQUESTED** [review]

**Comment by Rexben001** [comment]
Lastly, there are no tests for your feature

File: controllers/cart/getCart.js (hunk 2 above)
Comment (aapetsi) [reply controllers/cart/getCart.js:32]: yeah. this is a sample output. the contents are expanded into a property called `productObjects`
```
{
    "_id": "5e1e192390f5d236c6107b04",
    "contents": [
      "5e1defb857aff215deecc4ad",
      "5e1df13b57aff215deecc4ae",
      "5e1df14457aff215deecc4af"
    ],
    "checkedOut": false,
    "agreedPrice": 45,
    "total": 540,
    "storeId": "5e1def4257aff215deecc4ac",
    "checkoutDate": "2020-01-14T19:40:19.557Z",
    "__v": 0,
    "productObjects": [
      {
        "_id": "5e1defb857aff215deecc4ad",
        "images": [
          "mee.jpg",
          "us.jpg"
        ],
        "name": "Shoes",
        "description": "A very nice ",
        "price": 500,
        "stock": 10,
        "storeId": "5e1def4257aff215deecc4ac",
        "__v": 0
      },
      {
        "_id": "5e1df13b57aff215deecc4ae",
        "images": [
          "mee.jpg",
          "us.jpg"
        ],
        "name": "Shoes1",
        "description": "A very nice ",
        "price": 500,
        "stock": 10,
        "storeId": "5e1def4257aff215deecc4ac",
        "__v": 0
      },
      {
        "_id": "5e1df14457aff215deecc4af",
        "images": [
          "mee.jpg",
          "us.jpg"
        ],
        "name": "Shoes2",
        "description": "A very nice ",
        "price": 500,
        "stock": 10,
        "storeId": "5e1def4257aff215deecc4ac",
        "__v": 0
      }
    ]
  }
```

Comment (Rexben001) [reply controllers/cart/getCart.js:32]: I feel the `productObjects` should replace `contents`

File: models/cart.js [hunk 4]
```diff
@@ -6,10 +6,13 @@ const cartSchema = new mongoose.Schema({
...
     required: true
   },
-  contents: {
-    type: Array,
-    required: true
-  },
+  contents: [
+    {
+      type: mongoose.Schema.Types.ObjectId,
+      ref: 'store',
```
Comment (DimejiAre) [inline models/cart.js]: @tolls-3 Should this reference store or product?

**Review by tolls-3 — APPROVED** [review]
Reviewed and all looks good!

**Review by Rexben001 — APPROVED** [review]

File: routes/cartRouter.js [hunk 5]
```diff
@@ -1,11 +1,14 @@
...
 
 // @route POST api/store/:store_id/cart
 // @desc  Add items to cart
 // @access Public
 router.post('/:store_id/cart', addCart)
 
+// @route GET api/store/:store_id/cart
+// @desc  Get items from cart
+// @access Public
+router.get('/:cart_id/', getCart)
```
Comment (DimejiAre) [inline routes/cartRouter.js:12]: I think this route is a bit confusing /api/store/123(id)
Even if you are not using the store id, you can go with /api/store/cart/123(id)

File: models/cart.js (hunk 4 above)
Comment (aapetsi) [reply models/cart.js]: typo ... fixed it to reference product

**Pull request merged**